src/database/populate.py

```
@@ -51,13 +53,15 @@ async def _preprocess_csv(self) -> pd.DataFrame:
         :rtype: pd.DataFrame
         """
         data = pd.read_csv(self._csv_file_path)
-        data = data.drop_duplicates(subset=['names', 'date_x'], keep='first')
-        data['crew'] = data['crew'].fillna('Unknown')
-        data['genre'] = data['genre'].fillna('Unknown')
-        data['genre'] = data['genre'].str.replace('\u00A0', '', regex=True)
-        data['date_x'] = data['date_x'].str.strip()
-        data['date_x'] = pd.to_datetime(data['date_x'], format='%m/%d/%Y', errors='coerce')
-        data['date_x'] = data['date_x'].dt.date
+        data = data.drop_duplicates(subset=["names", "date_x"], keep="first")
+        data["crew"] = data["crew"].fillna("Unknown")
+        data["genre"] = data["genre"].fillna("Unknown")
+        data["genre"] = data["genre"].str.replace("\u00a0", "", regex=True)
+        data["date_x"] = data["date_x"].str.strip()
+        data["date_x"] = pd.to_datetime(
+            data["date_x"], format="%m/%d/%Y", errors="coerce"
+        )
+        data["date_x"] = data["date_x"].dt.date
```

You convert `date_x` to datetime with `errors='coerce'` and then assign `.dt.date`. That produces `NaT`/`None` for invalid dates; later you insert `date` into a non-nullable DB column. You need to handle rows with invalid dates (e.g., drop them, provide a fallback, or raise a clear error) before seeding. Consider filtering out rows where `date_x` is null after parsing, or logging/skipping them so the DB insert won't fail silently or produce bad rows.